client/lib.go

```diff
@@ -62,7 +62,7 @@ func interpretBody(res *http.Response, decodeTo interface{}) error {
 	case http.StatusInternalServerError:
 		return fmt.Errorf("%w: %s", ErrInternalServerError, errRes.Error)
 	case http.StatusNotFound:
-		return fmt.Errorf("%w: %s", ErrNotFound, errRes.Error)
+		return fmt.Errorf("%w: %s", ErrNotFound, res.Request.URL.String())
```

Comment:
Why was this changed? Is the error in the response not helpful? Not existent?
Is it always assured that `Request` or `URL` be nil and cause a panic?
All the other invalid status codes check for the error returned in the body and this one does not